e2e/testinfra/terraform/modules/service_account/service_account.tf

```diff
@@ -40,3 +41,16 @@ resource "google_project_iam_member" "gcp_sa_role" {
   member  = "serviceAccount:${google_service_account.gcp_sa.email}"
   project = data.google_project.project.id
 }
+
+// Create IAM bindings directly to the KSA(s) for BYOID
+resource "google_project_iam_member" "k8s_sa_role_rootsync" {
```

Comment:
Technically, this is not associated with the GSA. I would rather it to be outside of the service account module.

It also needs an additional workload identity pool of the fleet host project (cs-dev-hub).